manager/allocator/networkallocator/networkallocator.go

```diff
@@ -284,7 +284,8 @@ func (na *NetworkAllocator) IsTaskAllocated(t *api.Task) bool {
 }
 
 // IsServiceAllocated returns if the passed service has its network resources allocated or not.
-func (na *NetworkAllocator) IsServiceAllocated(s *api.Service) bool {
+// init bool indicates if the func is called during allocator initialization stage.
+func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, init bool) bool {
```

Comment:
On the same line, can you please change this to:

```
func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, flags... bool) bool {
    onInit := false
    if len(flags) > 0 {
        onInit = flags[0]
    }
    <snip>
    if .... {
        return na.portAllocator.isPortsAllocatedOnInit(s, onInit)
    }
}
````

This will spare the code reader from wondering what that `false` flag is all the places he reads the call to `isServiceAllocated(..., false)`, where the the flag is true only in two instances to address a very specific case (IMO, restarting a one manger cluster is not a common operation).